src/main/java/org/spongepowered/common/registry/builtin/stream/TropicalFishShapeStreamGenerator.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.registry.builtin.stream;
+
+import org.spongepowered.api.CatalogKey;
+import org.spongepowered.api.data.type.TropicalFishShape;
+import org.spongepowered.api.util.Tuple;
+import org.spongepowered.common.data.type.SpongeTropicalFishShape;
+
+import java.util.stream.Stream;
+
+public final class TropicalFishShapeStreamGenerator {
+    private TropicalFishShapeStreamGenerator() {
+    }
+
+    public static Stream<Tuple<TropicalFishShape, Integer>> stream() {
+        return Stream.of(
+            Tuple.of(new SpongeTropicalFishShape(CatalogKey.minecraft("kob"), false, 0), 0),
```

Comment:
I'm tempted to ask for these constants to be exposed in the `Constants` class so that Data Key Processors will be able to use them (if they're supported to be changed). Thoughts?

Reply:
So like move the stream generation to Constants or did you mean something else.

Reply:
Okay after sleeping on it I think I understand a bit more what you meant. There is actually a variety enum. But its private static so the third AT has come.

So I will change it so it corresponds to the enum. However since all the Tropical Fish data integer data value based, literally nothing prevents someone from using mixin to add to BODY_TEXTURES (Base Texture used to make the fish)
PATTERN_TEXTURES_A (Pattern textures used on smaller fish)
PATTERN_TEXTURES_B (Pattern textures used on larger fish) and creating a mod which adds fish that can exceed these values. Even vanilla's NBT (via /data merge) allows you to exceed these values.

Reply:
I don't want these outside these basically util classes unless the need arises at some point.

Reply:
I saw there was a TropicalFishEntity.Type enum (why didn't I see this earlier lol), so I changed to that, although it comes at the cost of one AT but you can use the easier method to just set the variation of the fish within data.